### PR TITLE
fixed #13215: Lyrics don't paste word by word anymore

### LINKED ISSUE #13215: [MU4 Issue] Regression - Lyrics don't paste word by word anymore
In MS3 I was able to use Ctrl-v repeatedly to paste lyrics one word at a time from the clipboard. This does not work in MS4 anymore and only the first word is pasted... repeatedly.
I tested this in the latest Alpha release

1. Create a score
2. Copy a paragraph of lyrics from a text editor to the clipboard
3. Use Ctrl+L to go into lyric entry mode
4. Use Ctrl+V repeatedly to paste each word associated with subsequent notes

**Expected behavior**
I expect each word in the copied text to paste one by one under subsequent notes in the score like they did in MS3

**Platform information**
 - OS: ArchLinux

**Additional context**
This is something I need for my workflow. I hope it gets added again in the final version.

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #13215

### CODE REVIEW

**Review by RomanPudashkin — APPROVED** [review]

**Review by Jojo-Schmitz — APPROVED** [review]
Works for me, on Windows 11, and the same was as in 3.6.2.

**Comment by DmitryArefiev** [comment]
Tested on Win10. I found the first word after each line break is missing:


https://github.com/musescore/MuseScore/assets/90187801/d56d7136-3b98-4a8f-a941-f93a7dc48902

**Comment by Jojo-Schmitz** [comment]
Doesn't work for @MarcSabatella on Linux, but doesn't with 3.6.2 for him there either.
I suspect this special casing for Linux in 
https://github.com/musescore/MuseScore/pull/17524/files#diff-26bd3810778f254978e70572c39acd61846cc6e935d6376e25848725b8597e3fL3624-R3644
to be the culprit and am testing that theory with https://github.com/Jojo-Schmitz/MuseScore/pull/113 against 3.x

**Comment by Jojo-Schmitz** [comment]
Marc confirmed that my change worked for him.
In Mu4 there currently are 3 more places with this special casing, all in notationinteraction.cpp, and Marc is having issues with pasting in other places too, not just lyrics.
As I'm not having a Linux to try this on, I'd rather leave that to others.

**Comment by Jojo-Schmitz** [comment]
But see (and check/test) #17527

**Comment by MarcSabatella** [comment]
Indeed, for me on my Linux systems, paste into a score from the system clipboard doesn't work at all - not in lyrics, not in staff text or titles, etc.  Only place I can paste from the system clipboard is into the UI itself - eg, dialog box text fields.  Conversely, if I copy *from* text in the score, I can only paste into other places in the score, not into anything that tries to access the system clipboard - dialog boxes or other programs.

I assume there must be some reason those ifdef's exist, and that *some* Linux systems work better with QClipboard::Selection, but I have no idea what the actual issues are with this.  I'll be checking the artifact from @Jojo-Schmitz 's test build when available.

File: src/notation/internal/notationinteraction.cpp [hunk 1]
```diff
@@ -3616,18 +3639,55 @@ void NotationInteraction::copyLyrics()
...
 
     if (isTextEditingStarted()) {
-        #if defined(Q_OS_MAC) || defined(Q_OS_WIN)
+#if defined(Q_OS_MAC) || defined(Q_OS_WIN)
         QClipboard::Mode mode = QClipboard::Clipboard;
-        #else
+#else
         QClipboard::Mode mode = QClipboard::Selection;
-        #endif
+#endif
```
Comment (Jojo-Schmitz) [inline src/notation/internal/notationinteraction.cpp:3644]: I still believe this stuff should go and only the `QClipboard::Mode mode = QClipboard::Clipboard;` should stay

Comment (Eism) [reply src/notation/internal/notationinteraction.cpp:3644]: Of course, this needs to be fixed in a separate PR, not here

Comment (Jojo-Schmitz) [reply src/notation/internal/notationinteraction.cpp:3644]: I see. I'll rebase my PR then, once this here is merged ;-)

Comment (Jojo-Schmitz) [reply src/notation/internal/notationinteraction.cpp:3644]: or you could cherry-pick mine

**Comment by Jojo-Schmitz** [comment]
rebase needed

**Comment by MarcSabatella** [comment]
And of course, the special casing of the clipboard by OS should be removed here as well now that it has been for normal paste as well.

**Comment by DmitryArefiev** [comment]
Tested #13215 on Win10, Mac13, LinuxUbuntu 22.04 - **FIXED**

**Comment by MarcSabatella** [comment]
Works great for me on Linux, thanks!

**Comment by Jojo-Schmitz** [comment]
For me on Windows 11 too

**Comment by DmitryArefiev** [comment]
Great! Thanks for testing!

**Pull request merged**